depthai_nodes/ml/parsers/segmentation.py

```diff
@@ -143,10 +143,11 @@ def run(self):
 
             class_map = (
                 np_function(segmentation_mask, axis=0)
-                .reshape(segmentation_mask.shape[1], segmentation_mask.shape[2], 1)
-                .astype(np.uint8)
+                .reshape(segmentation_mask.shape[1], segmentation_mask.shape[2])
+                .astype(np.int16)
             )
+            class_map = class_map - 1
```

Comment:
I think this is needed only for the case where `segmentation_mask.shape[0]==1` and we manually add a background class. Or even better what if we instead of np.zeros() there stack an array of -1s? Because for other cases we can't assume that class at "slice" 0 is background class. It could be that we have panoptic segmentation where every pixel is segmented into a distinct class and this might not be "background".